tests/Middleware/MiddlewareTestCase.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+/*
+ * This file is part of Raven.
+ *
+ * (c) Sentry Team
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sentry\Tests\Middleware;
+
+use PHPUnit\Framework\TestCase;
+use Psr\Http\Message\ServerRequestInterface;
+use Sentry\Configuration;
+use Sentry\Event;
+
+abstract class MiddlewareTestCase extends TestCase
+{
+    /**
+     * @param callable                    $middleware
+     * @param Event|null                  $event
+     * @param ServerRequestInterface|null $request
+     * @param \Exception|null             $exception
+     * @param array                       $payload
+     *
+     * @return Event The event returned by the middleware
+     */
+    protected function assertMiddlewareInvokesNext(
+        callable $middleware,
+        Event $event = null,
+        ServerRequestInterface $request = null,
+        \Exception $exception = null,
+        array $payload = []
+    ) {
+        $callbackInvoked = false;
+        $callback = function (Event $passedEvent, $passedRequest = null, $passedException = null) use (&$callbackInvoked) {
+            $this->assertInstanceOf(ServerRequestInterface::class, $passedRequest);
+
+            if ($passedException) {
+                $this->assertInstanceOf(\Exception::class, $passedException);
+            }
+
+            $callbackInvoked = true;
+
+            return $passedEvent;
+        };
+
+        if (null === $event) {
+            $event = new Event($this->createMock(Configuration::class));
+        }
+        if (null === $request) {
+            $request = $this->createMock(ServerRequestInterface::class);
+        }
+        if (null === $exception) {
+            $exception = new \Exception();
+        }
+
+        $returnedEvent = $middleware($event, $callback, $request, $exception, $payload);
+
+        $this->assertTrue($callbackInvoked, 'Next middleware was not invoked');
+        $this->assertSame($event, $returnedEvent);
```

Comment:
Should we check that `returnedEvent` is an instance of `Event` instead of asserting that it's the same instance of the event that has been passed to the middleware? I mean, the real purpose of the assert should be to verify that an event is returned, it makes no difference if the instance is the same as the previous one

Reply:
Since it's not an immutable object, it doesn't make sense to change the instance anywhere. In fact, even with this assertion, the tests never fail.